crates/vm/levm/src/vm.rs

```diff
@@ -602,7 +602,10 @@ impl VM {
 
         // 1. Undo value transfer if the transaction was reverted
         if let TxResult::Revert(_) = report.result {
-            self.decrease_account_balance(receiver_address, initial_call_frame.msg_value)?;
+            // msg_value was not increased in the receiver account when is a create transaction.
+            if !self.is_create() {
+                self.decrease_account_balance(receiver_address, initial_call_frame.msg_value)?;
+            }
```

Comment:
Same comment as in the other PR, I would've initialized contract with balance 0 and avoided that if statement, but again, is personal preference.